src/stream.rs

```diff
@@ -407,27 +410,38 @@ pub(crate) fn connect_host(
         stream.set_write_timeout(unit.agent.config.timeout_write)?;
     }
 
-    if proto == Some(Proto::HTTPConnect) {
+    if proto == Some(Proto::HTTPConnect) || proto == Some(Proto::HTTPSConnect) {
         if let Some(ref proxy) = proxy {
-            write!(stream, "{}", proxy.connect(hostname, port)).unwrap();
-            stream.flush()?;
+            let mut s;
+            if proto == Some(Proto::HTTPSConnect) {
+                s = unit
+                    .agent
+                    .config
+                    .tls_config
+                    .connect(&proxy.server, Box::new(stream))?;
+            } else {
+                s = Box::new(stream);
+            }
+            write!(s, "{}", proxy.connect(hostname, port)).unwrap();
```

Comment:
I see that we already have this `.unwrap()`, but it would be nice to cleanup while we're here to use a `?` instead.